stable/aws-for-fluent-bit/templates/configmap.yaml

```diff
@@ -205,11 +261,194 @@ data:
+        {{- end }}
+        {{- if .Values.s3.storeDir }}
+        store_dir                     {{ .Values.s3.storeDir }}
+        {{- end }}
+        {{- if .Values.s3.storeDirLimitSize }}
+        store_dir_limit_size          {{ .Values.s3.storeDirLimitSize }}
+        {{- end }}
+        {{- if .Values.s3.s3KeyFormat }}
+        s3_key_format                 {{ .Values.s3.s3KeyFormat }}
+        {{- end }}
+        {{- if .Values.s3.s3KeyFormatTagDelimiters }}
+        s3_key_format_tag_delimiters  {{ .Values.s3.s3KeyFormatTagDelimiters }}
+        {{- end }}
+        {{- if .Values.s3.staticFilePath }}
+        static_file_path              {{ .Values.s3.staticFilePath }}
+        {{- end }}
+        {{- if .Values.s3.usePutObject }}
+        use_put_object                {{ .Values.s3.usePutObject }}
+        {{- end }}
+        {{- if .Values.s3.roleArn }}
+        role_arn                      {{ .Values.s3.roleArn }}
+        {{- end }}
+        {{- if .Values.s3.endpoint }}
+        endpoint                      {{ .Values.s3.endpoint }}
+        {{- end }}
+        {{- if .Values.s3.stsEndpoint }}
+        sts_endpoint                  {{ .Values.s3.stsEndpoint }}
+        {{- end }}
+        {{- if .Values.s3.cannedAcl }}
+        canned_acl                    {{ .Values.s3.cannedAcl }}
+        {{- end }}
+        {{- if .Values.s3.compression }}
+        compression                   {{ .Values.s3.compression }}
+        {{- end }}
+        {{- if .Values.s3.contentType }}
+        content_type                  {{ .Values.s3.contentType }}
+        {{- end }}
+        {{- if .Values.s3.sendContentMd5 }}
+        send_content_md5              {{ .Values.s3.sendContentMd5 }}
+        {{- end }}
+        {{- if .Values.s3.autoRetryRequests }}
+        auto_retry_requests           {{ .Values.s3.autoRetryRequests }}
+        {{- end }}
+        {{- if .Values.s3.logKey }}
+        log_key                       {{ .Values.s3.logKey }}
+        {{- end }}
+        {{- if .Values.s3.preserveDataOrdering }}
+        preserve_data_ordering        {{ .Values.s3.preserveDataOrdering }}
+        {{- end }}
+        {{- if .Values.s3.storageClass }}
+        storage_class                 {{ .Values.s3.storageClass }}
+        {{- end }}
+        {{- if .Values.s3.retryLimit }}
+        retry_limit                   {{ .Values.s3.retryLimit }}
+        {{- end }}
+        {{- if .Values.s3.externalId }}
+        external_id                   {{ .Values.s3.externalId }}
+        {{- end }}
+        {{- if .Values.s3.extraOutputs }}
+{{ .Values.s3.extraOutputs | indent 8 }}
+        {{- end }}
+{{- end }}
+
+{{- if .Values.opensearch.enabled }}
+    [OUTPUT]
+        Name                opensearch
+        Match               {{ .Values.opensearch.match }}
+        AWS_Region          {{ .Values.opensearch.awsRegion }}
+        AWS_Auth            {{ .Values.opensearch.awsAuth }}
```

Comment:
The user technically doesn't have to enable these I think since you can run opensearch yourself without AWS Auth? 

I'm not sure, but just in case it might be best to wrap this like the other values so that if not specified they aren't rendered. Otherwise you've made region required.

Reply:
Done.

Reply:
It was taken from elastic search section though. Do we want to have similar implementation there as well?